File: src/main/webapp/static/js/admin/result.js
if(typeof Result == "undefined"){
    var Result = {
        Success: {
            code: 901,
            name: "Success",
            message: "Success"
        },
        SessionError: {
            code: 902,
            name: "SessionError",
            message: "Session is timeout, login again!"
        },
        ObjectIdError: {
            code: 903,
            name: "ObjectIdError",
            message: "Cannot find the object by this object id."
        },
        SaveInternalError: {
            code: 904,
            name: "SaveInternalError",
            message: "Cannot save the object due to a server internal error."
        },
        CategoryRemoveNotAllow: {
            code: 1111,
            name: "CategoryRemoveNotAllow",
            message: "Actived category or category with selections cannot be removed."
        },
        SelectionRemoveNotAllow: {
            code: 1211,
            name: "SelectionRemoveNotAllow",
            message: "Actived selection or selection with options cannot be removed."
        },
        OptionRemoveNotAllow: {
            code: 1311,
            name: "OptionRemoveNotAllow",
            message: "Actived option cannot be removed."
        }
    }
}

File: src/main/java/org/mushare/tsukuba/service/common/Result.java
package org.mushare.tsukuba.service.common;

import org.mushare.tsukuba.domain.Category;
import org.mushare.tsukuba.domain.User;

public enum Result {

    Success(901),
    SessionError(902),
    ObjectIdError(903),
    SaveInternalError(904),

    UserEmailRegistered(1001),
    CategoryRemoveNotAllow(1002),
    SelectionRemoveNotAllow(1003),
    OptionRemoveNotAllow(1004),
    MessageModifyNoPrevilege(1006),
    FavoriteNotExisted(1007);

    public int code;
    public Object object;

    private Result(int code) {
        this.code = code;
    }

    private Result(int code, Object object) {
        this.code = code;
        this.object = object;
    }

}
